builder/comp-builder.nix

```diff
@@ -346,6 +346,8 @@ let
           # so we are dealing with a sublib. As we build sublibs separately, the above
           # query should be safe.
           echo "--dependency=''${name#z-${package.identifier.name}-z-}=$id" >> $out/exactDep/configure-flags
+          # Allow package-name:sublib-name to work in build-depends
```

Comment:
Maybe it's obvious if you know how the package dbs work, but it's rather mysterious to me why this is the right thing to do. Maybe worth a longer comment?

Reply:
> Maybe it's obvious if you know how the package dbs work

Not really.  I just noticed that the argument that was passed did not include the package name and guessed that adding one the same where it was included might help.  I kept the comment short because I was worried mistakes in a longer explanation would be a nightmare to fix.  Changing anything here (even a typo in the comments) causes _everything_ to rebuild on hydra.

I'll refactor this code to use nix comments instead of bash ones and move some of the logic from bash to nix too (it is annoying that changing the code that is only used for sub lib components causes a full rebuild).  Then we can easily add a better explanation for why we think it works.

Reply:
That's a great idea about moving the logic into Nix where possible!

Reply:
It is quite silly to have to rebuild the world when you change a comment!